src/dc_receive_imf.rs

```diff
@@ -569,7 +569,7 @@ fn add_parts(
         *chat_id,
         *from_id,
         *sent_timestamp,
-        if 0 != flags & DC_IMAP_SEEN { 0 } else { 1 },
+        0 == flags & DC_IMAP_SEEN,
```

Comment:
took me a bit to be sure this is correct -- this is where i miss keyword-calling convention like in python. 
As we've had many boolean inversion errors so far it might be worthwhile to do:
```
let is_fresh = 0 == flags & DC_IMAP_SEEN;
calc_timestamps(
    ...
    is_fresh
    ...
)```